plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -89,12 +112,25 @@ func PredicateMetadata(pod *api.Pod, nodeInfoMap map[string]*schedulercache.Node
 	if err != nil {
 		return nil
 	}
-	return &predicateMetadata{
+
+	predicateMetadata := &predicateMetadata{
 		podBestEffort:             isPodBestEffort(pod),
 		podRequest:                getResourceRequest(pod),
 		podPorts:                  getUsedPorts(pod),
 		matchingAntiAffinityTerms: matchingTerms,
+		servicePods:               make(map[string]([]*api.Pod)),
```

Comment:
We should call this make only in precomputation functions - othewise if this is unused, we are loosing unnecessary memory allocation.